awx/ui_next/src/components/MultiSelect/usePFSelect.js

```diff
@@ -0,0 +1,52 @@
+import { useState, useEffect } from 'react';
+
+/*
+  Hook for using PatternFly's <Select> component. Guarantees object equality
+  between objects in `value` and the corresponding objects loaded as
+  `options` (based on matched id value).
+ */
+export default function usePFSelect(value, onChange) {
+  const [options, setOptions] = useState([]);
+  const [selections, setSelections] = useState([]);
+
+  useEffect(() => {
+    if (value !== selections && options.length) {
+      console.log(value, typeof value);
+      const syncedValue = value.map(item =>
+        options.find(i => i.id === item.id)
+      );
+      setSelections(syncedValue);
+    }
+    /* eslint-disable-next-line react-hooks/exhaustive-deps */
+  }, [value, options]);
+
+  const onSelect = (event, item) => {
+    if (selections.includes(item)) {
+      onChange(selections.filter(i => i !== item));
+    } else {
+      onChange(selections.concat(item));
+    }
+  };
+
+  return {
+    selections: options.length ? addToStringToObjects(selections) : [],
+    onSelect,
+    options,
+    setOptions: newOpts => setOptions(addToStringToObjects(newOpts)),
+  };
+}
+
+/*
+  PF uses toString to generate React keys. This is used to ensure
+  all objects in the array have a toString method.
+ */
+function addToStringToObjects(items = []) {
+  items.forEach(item => {
+    item.toString = toString;
+  });
+  return items;
+}
+
+function toString() {
+  return this.id;
```

Comment:
Should this return a string? In case someone working on this PF side of this decides to do something string-like with the return value of `value.toString`.